tests/unit/matrix/test_assembler.py

```diff
@@ -1561,11 +1562,26 @@ def it_provides_the_order(self, request):
             request, _SortRowsByMarginalHelper, "_empty_row_idxs"
         )
 
-        assert _SortRowsByMarginalHelper(None, None)._order
+        _SortRowsByMarginalHelper(None, None)._order
 
         SortByValueCollator_.display_order.assert_called_once_with(
             _rows_dimension_(),
             _element_values_(),
             _subtotal_values_(),
             _empty_row_idxs_(),
         )
+
+    def it_provides_the_element_values_to_help(self, request, _marginal_prop_):
+        assert _SortRowsByMarginalHelper(None, None)._element_values == "a"
+
+    def it_provides_the_subtotal_values_to_help(self, request, _marginal_prop_):
+        assert _SortRowsByMarginalHelper(None, None)._subtotal_values == "b"
+
+    # fixture components ---------------------------------------------
+
+    @pytest.fixture
+    def _marginal_prop_(self, request):
+        marginal_ = instance_mock(request, _BaseMarginal, blocks=["a", "b"])
+        return property_mock(
+            request, _SortRowsByMarginalHelper, "_marginal", return_value=marginal_
+        )
```

Comment:
I find it wiser to not make pytest fixtures anything more than a single, simple mock (no specified return or attribute values). I used to do things like this, but discovered after a while that it made the fixtures less flexible and also makes it so you have to scroll down to the fixtures to understand the test. That latter reason is probably the more compelling one.

So here you would have both `_marginal_prop_` and `marginal_` as fixtures, inject them both in the test, and then have `_marginal_prop_.return_value = marginal_` in the test fixturing block. Also, the `marginal_.blocks = ["a", "b"]` would be in the test fixturing block and you can make full sense of the test without ever scrolling down (which is a real pain when test suites get bigger).

So a test like this is easy to make sense of. Really the name should tell you everything you need to know about a mock:

```python
def it_provides_the_element_values_to_help(self _marginal_prop_, marginal_):
    _marginal_prop_.marginal_ = marginal_
    marginal_.blocks = ["a", "b"]

    assert _SortRowsByMarginalHelper(None, None)._element_values == "a"
```